### PR TITLE
Add ability to manage resource policy for AWS Secrets Manager secrets

### PR DESCRIPTION
**Pull request by ykrysko**
##### SUMMARY

AWS Secrets Manager secrets support attaching resource policy. The benefit is huge when necessary to access secrets from other AWS accounts. This pull request adds ability to manage (add new/remove or modify existing) secrets resource policy.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

module: aws_secret

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by ykrysko** [comment]
Added required IAM permissions to the aws-terminator repo, need to trigger retest

**Comment by ykrysko** [comment]
Submitted a pull request to modify IAM permissions to fix failing tests https://github.com/mattclay/aws-terminator/pull/183

**Comment by jillr** [comment]
recheck

**Comment by marknet15** [comment]
Hey @ykrysko I think you've just got the 1 last sanity check failure issue:
```
ERROR: Found 1 pep8 issue(s) which need to be resolved:
ERROR: plugins/modules/aws_secret.py:58:114: W291: trailing whitespace
```

**Comment by ykrysko** [comment]
@marknet15 , thanks! Fixed.

File: plugins/modules/aws_secret.py [hunk 1]
```diff
@@ -54,6 +53,13 @@
     - Specifies string or binary data that you want to encrypt and store in the new version of the secret.
     default: ""
     type: str
+  resource_policy:
+    description:
+    - Specifies JSON-formatted resource policy to attach to the secret. Useful when granting cross-account access
+       to secrets.
+    required: false
+    type: json
+    version_added: 2.2.0
```
Comment (marknet15) [inline plugins/modules/aws_secret.py:62]: @markuman would this be `2.3.0` now ?

**Comment by marknet15** [comment]
@ykrysko just noticed, but you also need to add a changelog fragment
https://github.com/ansible-collections/community.aws/tree/main/changelogs/fragments

**Comment by ykrysko** [comment]
> @ykrysko just noticed, but you also need to add a changelog fragment https://github.com/ansible-collections/community.aws/tree/main/changelogs/fragments

Thanks, @marknet15 . Added.

File: plugins/modules/aws_secret.py (hunk 1 above)
Comment (ykrysko) [reply plugins/modules/aws_secret.py:62]: if version_added means collection version, the next one is 3.0.1. @markuman, could you please advise when you get a chance? Thanks.

Comment (alinabuzachis) [reply plugins/modules/aws_secret.py:62]: I guess 3.0.1 has been released. I would probably specify 3.1.0 if that makes sense for @markuman.

Comment (markuman) [reply plugins/modules/aws_secret.py:62]: Oh yes. The release train is moving fast this year.  
+1 for 3.1.0

Comment (ykrysko) [reply plugins/modules/aws_secret.py]: Thank you, @markuman and @alinabuzachis . Bumped the version_added to 3.1.0

File: plugins/modules/aws_secret.py [hunk 2]
```diff
@@ -255,6 +301,15 @@ def delete_secret(self, name, recovery_window):
             self.module.fail_json_aws(e, msg="Failed to delete secret")
         return response
 
+    def delete_resource_policy(self, name):
+        if self.module.check_mode:
+            self.module.exit_json(changed=True)
+        try:
+            response = self.client.delete_resource_policy(SecretId=name)
+        except Exception as e:
```
Comment (alinabuzachis) [inline plugins/modules/aws_secret.py:309]:
```suggestion
       except (BotoCoreError, ClientError) as e:
```

File: plugins/modules/aws_secret.py [hunk 3]
```diff
@@ -227,13 +260,26 @@ def create_secret(self, secret):
...
             self.module.fail_json_aws(e, msg="Failed to update secret")
         return response
 
+    def put_resource_policy(self, secret):
+        if self.module.check_mode:
+            self.module.exit_json(changed=True)
+        try:
+            json.loads(secret.secret_resource_policy_args.get("ResourcePolicy"))
+        except (TypeError, ValueError) as e:
+            self.module.fail_json_aws(e, msg="Failed to parse resource policy as JSON")
```
Comment (alinabuzachis) [inline plugins/modules/aws_secret.py:275]:
```suggestion
            self.module.fail_json(e, msg="Failed to parse resource policy as JSON")
```

File: plugins/modules/aws_secret.py [hunk 4]
```diff
@@ -211,6 +235,15 @@ def get_secret(self, name):
             self.module.fail_json_aws(e, msg="Failed to describe secret")
         return secret
 
+    def get_resource_policy(self, name):
+        try:
+            resource_policy = self.client.get_resource_policy(SecretId=name)
+        except self.client.exceptions.ResourceNotFoundException:
+            resource_policy = None
+        except Exception as e:
```
Comment (alinabuzachis) [inline plugins/modules/aws_secret.py:243]:
```suggestion
        except (BotoCoreError, ClientError) as e:
```

**Review by alinabuzachis — CHANGES_REQUESTED** [review]
@ykrysko That looks great! Just a few nits!

File: plugins/modules/aws_secret.py (hunk 4 above)
Comment (ykrysko) [reply plugins/modules/aws_secret.py]: @alinabuzachis , thank you! Fixed

File: plugins/modules/aws_secret.py (hunk 3 above)
Comment (ykrysko) [reply plugins/modules/aws_secret.py]: @alinabuzachis , thank you! Fixed

File: plugins/modules/aws_secret.py (hunk 2 above)
Comment (ykrysko) [reply plugins/modules/aws_secret.py]: @alinabuzachis , thank you! Fixed

**Comment by marknet15** [comment]
recheck

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**